avocado/core/parser.py

```diff
@@ -80,4 +81,16 @@ def finish(self):
 
         Side effect: set the final value for attribute `args`.
         """
-        self.args = self.application.parse_args(namespace=self.args)
+        self.args, _ = self.application.parse_known_args(namespace=self.args)
+        if _:
+            for sub in self.application._subparsers._actions:
+                if sub.dest == 'subcommand':
+                    sub.choices[self.args.subcommand].print_usage()
+                    # Respecting the argparse exit code on unknown arg.
+                    # Using exit_codes.AVOCADO_JOB_FAIL here seems just wrong.
+                    sys.exit(2)
```

Comment:
@clebergnu my concern is that a different code here will differ from others argpase errors, like this:

    [apahim@hurricane ~]$ avocado
    usage: avocado [-h] [-v] [--config CONFIG_FILE]
                   {exec-path,run,sysinfo,list,multiplex,plugins,config,distro,vt-bootstrap}
                   ...
    avocado: error: too few arguments
    [apahim@hurricane ~]$ echo $?
    2